aten/src/ATen/native/quantized/cudnn/Conv.cpp

```diff
@@ -233,25 +281,51 @@ void raw_cudnn_convolution_forward_out(
       .setxDesc(getTensorDescriptor(input, 'x', key.input_alignment))
       .setyDesc(getTensorDescriptor(conv_output, 'y', key.output_alignment))
       .setwDesc(getTensorDescriptor(weight, 'w', key.weight_alignment))
       .setcDesc(getConvDescriptor(key.params.dataType, padding, stride, dilation))
       .build();
   // std::cout << "operator:" << conv_op.describe() << std::endl;
-  // TODO: add support for bias
+
+  c10::optional<cudnn_frontend::Operation> bias_mult_op;
+  c10::optional<cudnn_frontend::Operation> sum_conv_bias_op;
+  if (bias.has_value()) {
+    // we can't directly assign bias_mult_op becauase operator= is deleted for cudnn_frontend::Operation;
+    // alternatively, I think we can use std::unique_ptr and dynamically allocate these builder ops
+    // but here, we chose to do it statically. c10::optional<T>::emplace() enables this approach
+    // TODO: can we assign the result back into bias and get rid of after_scales_bias? pending NVIDIA response
+    bias_mult_op.emplace(cudnn_frontend::OperationBuilder(CUDNN_BACKEND_OPERATION_POINTWISE_DESCRIPTOR)
+      .setxDesc(getTensorDescriptor(broadcasted_bias.value(), 'b', getAlignment(broadcasted_bias.value())))
+      .setbDesc(getTensorDescriptor(bias_multiplier_tensor.value(), 'c', getAlignment(bias_multiplier_tensor.value())))
+      .setyDesc(getTensorDescriptor(after_scales_bias.value(), 'd', getAlignment(after_scales_bias.value())))
+      .setpwDesc(getPointWiseMulDescriptor(getCudnnDataType(bias_multiplier_tensor.value())))
+      .build());
+
+    // TODO: can we assign the result back into conv_output and get rid of after_add?
+    sum_conv_bias_op.emplace(cudnn_frontend::OperationBuilder(CUDNN_BACKEND_OPERATION_POINTWISE_DESCRIPTOR)
+      .setxDesc(conv_op.getOutputTensor())
+      .setbDesc(getTensorDescriptor(after_scales_bias.value(), 'd', getAlignment(after_scales_bias.value())))
+      .setyDesc(getTensorDescriptor(after_add.value(), 'e', getAlignment(after_add.value())))
+      .setpwDesc(getPointWiseAddDescriptor(getCudnnDataType(after_scales_bias.value())))
+      .build());
+  }
 
   auto requant_op = cudnn_frontend::OperationBuilder(CUDNN_BACKEND_OPERATION_POINTWISE_DESCRIPTOR)
-    .setxDesc(conv_op.getOutputTensor())
+    .setxDesc(bias.has_value() ? sum_conv_bias_op.value().getOutputTensor() : conv_op.getOutputTensor())
     .setbDesc(getTensorDescriptor(requantize_multiplier_tensor, 's', getAlignment(requantize_multiplier_tensor)))
     .setyDesc(getTensorDescriptor(quantized_output.sizes(), quantized_output.strides(), CUDNN_DATA_INT8, 'r', getAlignment(quantized_output)))
     .setpwDesc(getPointWiseMulDescriptor(getCudnnDataType(requantize_multiplier_tensor)))
     .build();
   // std::cout << "operator:" << requant_op.describe() << std::endl;
 
-  std::array<cudnn_frontend::Operation const *, 2> ops = {&conv_op, &requant_op};
+  std::vector<cudnn_frontend::Operation const *> ops{bias.has_value()};
```

Comment:
where is conv_op? why do we have `bias.has_value()` here? is this supposed to be `&conv_op`?

Reply:
yes, sorry, I forgot to ghstack the latest amended commit that fixed this. I'll get that up right away